File: 001/cpp/src/main.cpp
//
// main.cpp
// Project Euler - Problem 1
//
// Jordan Sola 2020 - MIT License

#include <iostream>

using std::cout;
using std::endl;

int main()
{
    int limit = 1000;
    int sum = 0;

    // ----- multiples
    int a = 3;
    int b = 5;
    // -----

    for (int i = 0; i < limit; i++)
    {
        if ((i % a == 0) || (i % b == 0))
        {
            sum += i;
        }
    }

    cout << sum << endl;

    return 0;
}

File: 001/javascript/main.js
//
// main.js
// Project Euler - Problem 1
//
// Jordan Sola 2021 - MIT License

/*
Given a limit, find the sum of all the
factors of 3 and 5 up to the limit.
*/

var limit = 1000;
var sum = 0;

var factor_a = 3;
var factor_b = 5;

for (var i = 0; i < limit; i++)
{
    if (i % factor_a == 0|| i % factor_b == 0)
    {
        sum += i;
    }
}

WScript.echo(sum);